lib/puppet/pops/types/types.rb

```diff
@@ -378,12 +426,17 @@ def initialize(values)
 
       # Returns Enumerator if no block is given, otherwise, calls the given
       # block with each of the strings for this enum
-      def each
-        if block_given?
-          values.each { |x| yield x }
-        else
-          values.to_enum
-        end
+      def each(&block)
+        r = Iterable.on(self)
+        block_given? ? r.each(&block) : r
+      end
+
+      def iterable?
+        true
+      end
+
+      def iterable_type
+        PIterableType.new(PStringType::DEFAULT)
```

Comment:
Doesn't this lose precision? For an Enum, the strings are known. so it is possible to do either Iterable[String[min,max]], or why not just Iterable[<this enum>] ?